pylib/openrazer/client/devices/mice.py

```diff
@@ -97,6 +97,75 @@ def dpi(self, value: tuple):
         else:
             raise NotImplementedError()
 
+    @property
+    def dpi_stages(self) -> list:
+        """
+        Get mouse DPI stages
+
+        Will return a list of tuples
+        :return: DPI stages [(500, 500), (1000, 1000), (2000, 2000) ...]
+        :rtype: list
+
+        :raises NotImplementedError: if function is not supported
+        """
+        if self.has('dpi'):
+            raw_dpi_stages = self._dbus_interfaces['dpi'].getDPIStages()
+            dpi_stages = []
+
+            i = 0
+            while i + 1 < len(raw_dpi_stages):
+                dpi_x, dpi_y = raw_dpi_stages[i:i + 2]
+                i += 2
+
+                dpi_stages.append((int(dpi_x), int(dpi_y)))
+
+            return dpi_stages
+
+    @dpi_stages.setter
+    def dpi_stages(self, value: list):
+        """
+        Set mouse DPI stages
+
+        Daemon does type validation but can't be too careful
+        :param value: list of DPI X, Y tuple
+        :type value: list
+
+        :raises ValueError: If a tuple isn't long enough or contains invalid
+                            crap
+        :raises NotImplementedError: If function is not supported
+        """
+        if self.has('dpi'):
+            dpi_stages = []
+
+            for stage in value:
+                if len(stage) != 2:
+                    raise ValueError(
+                        "DPI tuple is not of length 2. Length: {0}".format(
+                            len(stage)))
+
+                dpi_x, dpi_y = stage
+
+                if not isinstance(dpi_x, int) or not isinstance(dpi_y, int):
+                    raise ValueError(
+                        "DPI X or Y is not an integer, X:{0} Y:{1}".format(
+                            type(dpi_x), type(dpi_y)))
+
+                if dpi_x < 0 or dpi_x > 16000:  # TODO add in max dpi option
```

Comment:
max dpi should be queryable from the device I think

Reply:
Done in 349c09d8dd9dc28676590aed51dc3ef5e5fb41ae